engine/function/src/templates/Basic.ftl

```diff
@@ -760,9 +757,18 @@ public class Basic {
 
         @SuppressWarnings("unchecked") final T[] result = (T[])Array.newInstance(trueCase.getComponentType(), n_c);
 
-        for (int i=0; i < n_c; i++) {
-            result[i] = condition.get(i) == null ? null : (condition.get(i) ? trueCase.get(i) : falseCase.get(i));
-        }
+        final Iterator<Boolean> ci = condition.iterator();
+        final Iterator<T> ti = trueCase.iterator();
+        final Iterator<T> fi = falseCase.iterator();
+        int i = 0;
+
+        while(ci.hasNext() && ti.hasNext() && fi.hasNext()) {
```

Comment:
Since you've established that the sizes are equal, I would just check `ci.hasNext()` and skip the others.